src/oc/hoc.cpp

```diff
@@ -73,17 +70,20 @@ int matherr1(void) {
 		fprintf(stderr, "Floating exception: Overflow\n");
 		break;
 	}
-#endif /*FE_NOMASK_ENV*/
 }
 #endif
 
+int nrn_feenableexcept_ = 0; // 1 if feenableexcept(FEEXCEPT) is successful
+
 void nrn_feenableexcept() {
   int result = -1;
-#if NRN_FLOAT_EXCEPTION && defined(FE_NOMASK_ENV)
+  nrn_feenableexcept_ = 0;
+#if NRN_FLOAT_EXCEPTION
   if (ifarg(1) && chkarg(1, 0., 1.) == 0.) {
     result = feenableexcept(0);
   }else{
     result = feenableexcept(FEEXCEPT);
+    nrn_feenableexcept_ = result ? 0 : 1;
```

Comment:
If `nrn_feenable_except` is called twice, wouldn't then on the second call `feenableexcept` return the `FEEXCEPT` mask in result? In this case I think this case would then set `nrn_feenableexcept_` to 0, which would be wrong.

https://linux.die.net/man/3/feenableexcept

I think we should always check that `feenableexcept` did not return -1 and set `nrn_feenableexcept_` depending on the input.

Reply:
So. ```feenableexcept(0)``` does not turn off SIGFPE, needed ```fedisableexcept(FEEXCEPT)```.
And need to make clear in the docs that the a successful return value of ```nrn_feenableexcept(...)``` is the previous floating-point mask. And on failure or not supported, -1.